src/Events/Migration/v0.24/03-setup-function.sql

```diff
@@ -0,0 +1,19 @@
+CREATE OR REPLACE FUNCTION events.getsubscriptions(
+	sourcehashset character varying[],
+	subject character varying,
+	type character varying)
+    RETURNS TABLE(id bigint, sourcefilter character varying, subjectfilter character varying, typefilter character varying, consumer character varying, endpointurl character varying, createdby character varying, validated boolean, "time" timestamp with time zone) 
+LANGUAGE 'plpgsql'
+AS $BODY$
+
+BEGIN
+return query 
+	SELECT s.id, s.sourcefilter, s.subjectfilter, s.typefilter, s.consumer, s.endpointurl, s.createdby, s.validated, s."time"
+	FROM events.subscription s
+  WHERE  s.sourcefilterhash = ANY(sourcehashset)
+  AND (s.subjectfilter is NULL OR length(subject) = 0 OR s.subjectfilter = subject)
```

Comment:
do we need to check for null if we're always sending in an empty string from the repository service?

Reply:
We don't always send in an empty string as far as I can tell. It should be possible to define a subject on a generic cloud event subscription. Will add a k6 test to verify that the logic is correct.

Reply:
Reevaluated the insert_subscription codepath and concluded that the existing `events.getsubscriptions` implementation will continue to work now that we always send a non-null value for `subject`. 
Test cases where `subject` is empty string:
subscriptionfilter is null --> return
subscriptionfilter is empty string --> return
subscriptionfilter length > 0 --> do not return

Only thing I don't like is that we don't trim whitespace on string properties in our request models.